src/pages/userPage/ActiveMembership/ActiveMembership.tsx

```diff
@@ -58,30 +56,39 @@ const ActiveMembership = () => {
     });
   };
   const getAppropriateToGenderDegree = (plastDegreeName: string): string => {
-    if (userGenders[0] === data?.user.gender?.name && plastDegreeName?.includes("/")) {
+    if (userGenders[0] === userProfile?.user.gender?.name && plastDegreeName?.includes("/")) {
       return plastDegreeName.split("/")[0];
-    } else if (userGenders[1] === data?.user.gender?.name && plastDegreeName?.includes("/")) {
+    } else if (userGenders[1] === userProfile?.user.gender?.name && plastDegreeName?.includes("/")) {
       return plastDegreeName.split("/")[1];
     } else return plastDegreeName;
   };
 
-
-  const fetchData = async () => {
-    if(UpdateData) UpdateData();
+  const InitialFetchData = async() => {
     const token = AuthStore.getToken() as string;
     setUserToken(jwt(token));
 
-    await userApi.getById(activeUserId).then(async (response) => {
-      setCurrentUser(response.data.user);
-    }).catch((error) => {
-      notificationLogic("error", error.message);
+    setAccessLevels(await activeMembershipApi.getAccessLevelById(userId));
+
+    await activeMembershipApi.getUserDates(userId).then((response) => {
+      response.dateEntry =
+        response.dateEntry === "0001-01-01T00:00:00" ? "" : response.dateEntry;
+      response.dateOath =
+        response.dateOath === "0001-01-01T00:00:00" ? "" : response.dateOath;
+      response.dateEnd =
+        response.dateEnd === "0001-01-01T00:00:00" ? "" : response.dateEnd;
```

Comment:
you have the magic string "0001-01-01T00:00:00", which repeats 3 times. It would be better to create the const variable for this string

Reply:
Thank you for great advice! :)